src/domain/alerts/alerts.repository.ts

```diff
@@ -51,33 +52,69 @@ export class AlertsRepository implements IAlertsRepository {
       return this._notifyUnknownTransaction(chainId, log);
     }
 
-    const safe = await this.safeRepository.getSafe({
-      chainId,
-      address: decodedEvent.args.to,
+    try {
+      const safeAddress = decodedEvent.args.to;
+
+      const newSafe = await this.predictNewSafeSetup({
+        chainId,
+        safeAddress,
+        decodedTransactions,
+      });
+
+      return this._notifyNewSafeSetup({ chainId, safeAddress, newSafe });
+    } catch {
+      return this._notifyUnknownTransaction(chainId, log);
+    }
+  }
+
+  private decodeTransactionAdded(data: Hex) {
+    try {
+      const decoded = this.safeDecoder.decodeFunctionData({ data });
+
+      if (decoded.functionName !== 'execTransaction') {
+        return [decoded];
+      }
+
+      return this.multiSendDecoder
+        .mapMultiSendTransactions(decoded.args[2])
```

Comment:
nitpick: I would extract `decoded.args[2]` so that it's more clear on what it represents.

Reply:
Extracted in ab0c720fca0ecfeba81054f22fc9ad347c96765d.